### ISSUE TITLE
[Timeline] Set a stacking limit

### ISSUE THREAD

**Issue by Tobi94** [almende/vis#3426]
It would be nice if we could set the maximum number of stacked items. Lets say we have 5 Items that would be stacked. But we set the limit to 3 Items. So only 3 Items get stacked and the remaining two are just overlapping and dont get stacked.

It would be nice if someone could point me in the right direction in case this is already possible. Alternatively it would be enough for me to just hide Items if a stack gets bigger than a specified size, so if the stack is bigger than x -> hide the stack

**Comment by yotamberk** [comment]
Awesome feature! I like it

**Comment by Tobi94** [comment]
I was able to archieve this myself with a small change in the stack function. 
![ezgif-4-839e00beff](https://user-images.githubusercontent.com/9142542/30387377-8f043450-98ac-11e7-9cd3-84e56561056e.gif)
Its not perfect but it works for me. In case anyone faces a similar problem, here is what my stack function now looks like. I just expanded the if condition, so that items only get stacked if the top-position of the lowest Item in the stack is below a certain threshold. In my case its 40. This obviously depends on the height of your items.


```
 exports.stack = function (items, margin, force) {
    if (force) {
      // reset top position of all items
      for (var i = 0; i < items.length; i++) {
        items[i].top = null;
      }
    }


    // calculate new, non-overlapping positions
    for (var i = 0; i < items.length; i++) {
      var item = items[i];
      if (item.stack && item.top === null) {
        // initialize top position
        item.top = 5;

        do {
          // TODO: optimize checking for overlap. when there is a gap without items,
          //       you only need to check for items from the next item on, not from zero
          var collidingItem = null;
          for (var j = 0, jj = items.length; j < jj; j++) {
            var other = items[j];
            if (item.top < 40 && other.top !== null && other !== item && other.stack && exports.collision(item, other, margin.item, other.options.rtl)) {
              collidingItem = other;
              break;
            }
          }

          if (collidingItem != null) {
            // There is a collision. Reposition the items above the colliding element
            item.top = collidingItem.top + collidingItem.height + margin.item.vertical;
           


          }
        } while (collidingItem);
      }
    }
  };
```

**Comment by mikeyc7m** [comment]
i tried this, but i just got the lower lines with one item each and the uppermost line mashed with all the items. 

it would be nicer if the extra items would simply "disappear" until you zoomed in enough to see them? or maybe they merged into a multi-event until you zoomed in to see the individual events?

(yeah, all ideas but no code, sorry)

**Comment by Tobi94** [comment]
Yeah this would be perfect. I'm not that good with javascript, but I was able to improve my solution.
Items now disappear if the stack gets to big. But I like your multi-event idea, would really love this to be a feature
![giphy](https://user-images.githubusercontent.com/9142542/30691609-93d84364-9ec8-11e7-926f-e2e4680c39ee.gif)

**Comment by yotamberk** [comment]
I think you're talking about clustering. Clustering has been addressed here: #242

**Comment by mikeyc7m** [comment]
clustering is awesome! but there's no documentation on it. how do we turn it on?

**Comment by mikeyc7m** [comment]
perhaps i've misunderstood, there is no clustering? 
instead i've come up with an alternative solution - when a selected item in the stack is above the visible viewport, zoom in until the viewport can fit it inside.

**Comment by Luiz-N** [comment]
+1 for a clustering solution!
